docs/spdx-tag-value-mapping.md

```diff
@@ -0,0 +1,151 @@
+2.8 | Creator         | **TBD**                   | could be something like `Creator: Tool: tern-XYZ`, with `XYZ` being Tern's version or commit hash
+2.9 | Created         | N/A                       | generate at document creation time
+
+### Packages
+
+#### Tern `Package`
+
+An SPDX "Package" maps fairly well to Tern's `Package` class. 
+
+ §  | SPDX Field name  | Tern data model reference | Comments
+----|------------------|---------------------------|---------
+3.1 | Package Name     | `Package.name()`          |
+3.2 | Package SPDX Identifier | **TBD**            | initially, probably use incrementing counter (e.g., `SPDXRef-Package-1`, `SPDXRef-Package-2` ...). May want to figure out a way to incorporate package name / version if we can de-duplicate to ensure uniqueness.
+3.3 | Package Version  | `Package.version()`       |
+3.7 | Package Download Location | **TBD**          | does Tern collect download location information? is there anything here that can be retrieved from package managers? if not, then use `NOASSERTION`
+3.8 | Files Analyzed   | N/A                       | will always be `FilesAnalyzed: false` unless we want to report on file-level analysis
+3.10| Package Checksum | **TBD**                   | appears that we don't collect any checksum info, at the Tern `Package` level. can we collect anything relevant for e.g. `apt-get` retrieved packages?
+3.13| Concluded License| **TBD**                   | to be discussed whether this should be `NOASSERTION` or should be same as Declared License
+3.15| Declared License | `Package.license()`       |
+3.17| Copyright Text   | **TBD**                   | does Tern collect copyright information? is there anything here that can be retrieved from package managers? if not, then use `NOASSERTION`
+3.21| External Reference | **TBD**                 | to be discussed whether to report on package manager identifiers in some fashion; also to discuss limited scope of existing SPDX package manager definitions
+
+#### Tern `ImageLayer`
+
+Note that a Tern `ImageLayer` could also be represented by an SPDX "Package". By doing so, Relationships can then be used to express the interactions between the `ImageLayer` and its packages.
+
+ §  | SPDX Field name  | Tern data model reference | Comments
+----|------------------|---------------------------|---------
+3.1 | Package Name     | `ImageLayer.diff_id()` or `ImageLayer.fs_hash()` | which of these hashes would be more appropriate? I assume there isn't another human-readable name for an `ImageLayer`
+3.2 | Package SPDX Identifier | **TBD**            | probably use something like `SPDXRef-ImageLayer-HASH` with HASH being either `ImageLayer.diff_id()` or `ImageLayer.fs_hash()`
+3.3 | Package Version  | N/A                       | likely omit? this is not a mandatory field and might not make sense for an `ImageLayer`
```

Comment:
It can -- `PackageVersion` can be any freeform text. SPDX doesn't prescribe a format for versions to be specified in.

However, if we were _always_ using a hash rather than a version number, then I might suggest omitting `PackageVersion` and using `PackageChecksum` instead. In either case the recipient of the SPDX document could make whatever use of the fields they chose, but `PackageChecksum` might be more useful if it'll always be a hash.

Reply:
I just saw your comment below in the `PackageChecksum` section -- makes sense and I think you can disregard what I just wrote in this thread  :)